File: CPP/BOJ-S3-9996-Connect_to_Server_When_Missing_Korea.cpp
#include <bits/stdc++.h>
using namespace std;

int n;
string s, pat, pre, suf;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  // freopen("input.txt", "r", stdin);
  // freopen("output.txt", "w", stdout);

  cin >> n;
  cin >> pat;
  int pos = pat.find('*');
  pre = pat.substr(0, pos);
  suf = pat.substr(pos + 1);

  for (int i = 0; i < n; ++i)
  {
    cin >> s;
    // 접두 + 접미 길이가 확인하려는 문자열보다 긴 경우
    if (pre.size() + suf.size() > s.size())
      cout << "NE\n";
    else
    {
      if (pre == s.substr(0, pre.size()) && suf == s.substr(s.size() - suf.size()))
        cout << "DA\n";
      else
        cout << "NE\n";
    }
  }

  return 0;
}

File: JS/BOJ-S3-9996-Connect_to_Server_When_Missing_Korea.js
"use strict";

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.trimEnd());

let output = [];
const solution = (input) => {
  const pat = input[1];
  const strings = input.slice(2);
  const pos = pat.indexOf('*');
  const pre = pat.substring(0, pos);
  const suf = pat.substring(pos + 1);
  
  strings.forEach(el => {
    if(el.length < pre.length + suf.length) output.push("NE");
    else {
      if(pre === el.substring(0, pre.length) && suf === el.substring(el.length - suf.length)) output.push("DA");
      else output.push("NE")
    }
  });
};

solution(input);
fs.writeFileSync("/dev/stdout", output.join("\n").trimEnd());
// fs.writeFileSync("../output.txt", output.join("\n").trimEnd(), (err) => console.error);
